aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -1008,4 +1008,47 @@ public GetObjectRequest withS3ObjectId(S3ObjectId s3ObjectId) {
         setS3ObjectId(s3ObjectId);
         return this;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
```

Comment:
I think wars have been fought over this topic already so I don't really want to go into the pros/cons of each approach. 😃 Josh Bloch (Effective Java) recommends using `instanceof` ([see here](http://www.artima.com/intv/bloch17.html)) and it's what the rest of the SDK does.

If we make the `equals`/`hashCode` methods final then we somewhat mitigate the issue because we can preserve the `a.equals(b) == b.equals(a)` behaviour even across subclasses.

`instanceof` is consistent with the rest of the SDK so I'd prefer to stick to that for now as it's less surprising.